test/lit/passes/optimize-instructions.wast

```diff
@@ -12188,6 +12188,33 @@
       )
     )
   )
+  ;; CHECK:      (func $ternary-identical-arms-return-select (param $x i32) (param $y i32) (param $z i32) (result i32)
+  ;; CHECK-NEXT:  (block $block
+  ;; CHECK-NEXT:   (select
+  ;; CHECK-NEXT:    (return
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (return
+  ;; CHECK-NEXT:     (local.get $y)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (local.get $z)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $ternary-identical-arms-return-select (param $x i32) (param $y i32) (param $z i32) (result i32)
+    (block $block
+      ;; we cannot optimize a select currently as the return has side effects
```

Comment:
What if it's a select of two blocks that end in unreachable? Not sure whether that's worth testing, though. It's definitely getting into the weeds :)

Reply:
We don't do this optimization if the children of the select are control flow structures like blocks (we'd need to handle branches etc. to the block, and other stuff). There is a test for that:

https://github.com/WebAssembly/binaryen/blob/375be0495e5e4e80cdd389bfe70bb16c1cbeccf8/test/lit/passes/optimize-instructions.wast#L12049-L12079